File: CrossDomainPGELLAv1.0/initPGELLA.m
function [model]=initPGELLA(Tasks,k,mu1,mu2,learningRateL)

model.T = 0;
model.S = zeros(k,size(Tasks,2));
model.mu_one = mu1;
model.mu_two = mu2; 
model.learningRateL = learningRateL;

% Determine the number of Groups
for i = 1:size(Tasks,2)
    GroupsArray(i) = Tasks(i).param.Group;
end

[Value,Idx]=max(GroupsArray);

for l = 1:Value
    for z = 1:size(Tasks,2)
        if Tasks(z).param.Group == l
            model(l).L = rand(Tasks(z).param.N*Tasks(z).param.M,k); % Initial L for each group
            break
        end
    end
end

File: src/vertical_control/scripts/pg_ella/initPGELLA.py
# -*- coding: utf-8 -*-
"""
Created on Tue Aug 18 01:39:30 2015

@author: brownyoda
"""

import numpy as np
from structs import Model
import sys

def initPGELLA(Tasks, k, mu_one, mu_two, learningRate):

    model = Model()
    model.S = np.zeros((k, np.shape(Tasks)[0]))
    #print model.S[:, 0]
    #print np.shape(model.S)
    #sys.exit(1)
    #model.S = [] * np.shape(Tasks)[0]
    model.mu_one = mu_one
    model.mu_two = mu_two
    model.learningRate = learningRate

    model.L = np.random.rand(Tasks[0].param.N * Tasks[0].param.M, k)

    return model